src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -344,7 +349,14 @@ object ScalafixPlugin extends AutoPlugin {
       if (shellArgs.rules.isEmpty && shellArgs.extra == List("--help")) {
         scalafixHelp
       } else {
-        val scalafixConf = scalafixConfig.in(config).value.map(_.toPath)
+        val scalafixConf =
+          if (scalafixRunExplicitly.value) scalafixConfig.in(config).value
+          else
+            scalafixOnCompileConfig
+              .in(config)
+              .?
+              .value
+              .orElse(scalafixConfig.in(config).value)
```

Comment:
I think you can also do
```suggestion
            scalafixConfig
              .in(config)
              .or(scalafixOnCompileConfig.in(config)(Some.apply))
              .value
```
Not sure if it's more readable though.

Reply:
(I think I mixed up the order but you get the point)

Reply:
Thank you for following up!
It's good to know the way you showed, and I would prefer yours since I feel the `.?` line a bit ugly 😅

Reply:
Hmm, `scalafixOnCompileConfig.in(config).apply(Option.apply)` returns `Initalize[Option[File]]` which doesn't have `or` operator.
So it seems to require casting to `SettingKey[Option[File]]`...🤔 

```scala
scalafixOnCompileConfig
   .in(config)(Option.apply)
   .or(scalafixConfig.in(config)).value
```

```
value or is not a member of sbt.Def.Initialize[Option[sbt.File]]
[error] possible cause: maybe a semicolon is missing before `value or'?
[error]               .or(scalafixConfig.in(config)).value
[error]                ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
```

Anyway, `or` operator implementation uses `?` so that it is no big difference in semantics. So please let me go with current `.?` way.

Reply:
Sure, apologies for the non-working proposal. Note that such a fallback won't be needed if we end up validating the other implementation https://github.com/scalacenter/sbt-scalafix/pull/148#discussion_r453842508, where `?` will be needed to conditionally pass an explicit config.

Reply:
@bjaglin No problem at all. Your suggestion helps my understanding a lot.

I've also confirmed https://github.com/scalacenter/sbt-scalafix/pull/148#discussion_r453842508 is working well and no need for `?` or `or` operators on this 👍